src/ports/postgres/modules/deep_learning/madlib_keras.py_in

```diff
@@ -544,6 +539,15 @@ def fit_final(state, **kwargs):
     return madlib_keras_serializer.serialize_weights_merge(
         loss, accuracy, image_count, weights)
 
+def get_segments_and_gpus(gpus_per_host):
+    gpus_per_host = 0 if gpus_per_host is None else gpus_per_host
+    segments_per_host = get_segments_per_host()
+
+    if 0 < gpus_per_host < segments_per_host:
+        plpy.warning('The number of gpus per host is less than the number of '
+                     'segments per host. The support for this case is '
+                     'experimental and it may fail.')
```

Comment:
Nothing is returned here, but I guess the calling functions expect `gpus_per_host` and `segments_per_host` to be returned right?

Reply:
Oh, yes forgot to test!  Fixing